src/generators/gencs.ml

```diff
@@ -2475,7 +2475,7 @@ let generate con =
 			(try
 				if cl.cl_interface then raise Not_found;
 				let cf = PMap.find "finalize" cl.cl_fields in
-				(if List.exists (fun c -> c.cf_name = "finalize") cl.cl_overrides then raise Not_found);
+				(if List.exists (fun c -> c.cf_name = "finalize" && has_class_field_flag c CfOverride) cl.cl_ordered_fields then raise Not_found);
```

Comment:
No need for `List.exists (fun c -> c.cf_name = "finalize"` anymore as it's already "checked" by `PMap.find "finalize"` in the previous line.

Reply:
I was worried about overloads, but I guess even in that case there's only one in `cl_ordered_fields`.

Reply:
Yes, AFAIR overloads are not pushed to `cl_ordered_fields`

Reply:
From the typer yes, but I wasn't sure what dark arts gencommon might be performing.

Anyway, looks like this is fine.